src/tm2/tm-device.cpp

```diff
@@ -1207,13 +1210,29 @@ namespace librealsense
             LOG_WARNING("Dropped frame. No valid profile");
             return;
         }
+
+        //Global base time sync may happen between two frames
+        //Make sure 2nd frame global timestamp is not impacted.
+        uint64_t global_ts = ts.global_ts.count();
+        uint32_t allowed_delta = 1000;
+        if (last_exposure != 0 && last_exposure < allowed_delta)
+            allowed_delta = last_exposure;
+
+        if (std::abs((int64_t)(message->rawStreamHeader.llNanoseconds - last_hw_ts)) < (int64_t)allowed_delta)
+        {
+            global_ts = last_global_ts - last_hw_ts + message->rawStreamHeader.llNanoseconds;
```

Comment:
@cchen6, at this point we have already applied `device_to_host_ns` in `get_coordinated_timestamp`.  Why are recreating it )or applying it again?) with `last_global_ts - last_hw_ts`?

Reply:
If device_to_host_ns is updated between two mono frames of stereo pair, the timestamp of two mono frames will be different causing mismatch while hw_ts is same for them. Adjust global_ts to be matched to hw_ts difference.